### DISCUSSION TITLE
Sub navigation with fetchContentNavigation()

### DISCUSSION THREAD

**Discussion by lucpotage** [nuxt/content#1156] [Q&A]
Hi! I have the following content structure:

```
content
- articles
- - article-1
- - article-2
- pages
- - page-1
- - page-2
```

I want to fetch the navigation of all articles and looked at the example here: [fetchContentNavigation()](https://content.nuxtjs.org/api/composables/fetch-content-navigation). However, it does not seem I can get the navigation of a sub directory.

I tried this without success:
```vue
<script setup>
const { data: navigation } = await useAsyncData('navigation', () => {
  return fetchContentNavigation('/articles')
})
</script>
```

I have the following error:
```
queryBuilder?.params is not a function
```

I see in the code base that this is supported.

Thanks for your help.

**Comment by lucpotage** [comment]
OK, I can do it with the `queryContent` function as follows:

```vue
<script setup>
const { data: navigation } = await useAsyncData('navigation', () => {
  return fetchContentNavigation(queryContent('/articles'))
})
</script>
```

But I get a filtered list rather than the sub navigation.

**Comment by harlan-zw** [comment]
Related PR for the error https://github.com/nuxt/content/discussions/1156

You may need to access the children from navigation

```vue
<template>
<ContentNavigation v-slot="{ navigation }" :query="catsQuery">
  <NuxtLink
    v-for="link of navigation[0].children"
    :key="link._path"
    :to="link._path"
  >
    {{ link.navTitle || link.title }}
  </NuxtLink>
</ContentNavigation>
</template>
```

**Comment by lucpotage** [comment]
While your solution works, it quickly becomes error prone with new content added.
Let's say I add another subdirectory in the content directory. My previous navigation item may not be at the first position anymore. This means that I would have to review that part of the code while there was no functional impact.

Besides, if the directory structure has multiple nested folders, it becomes even more fragile to maintain for bigger projects.

Having a way to only get the current navigation based on the queryContent result would help. As an example, when querying `fetchContentNavigation(queryContent('/articles'))`, I would expect the inner articles hierarchy as follows:

```
- article-1
- article-2
```

**Comment by itshizhan** [comment]
the same question.
and I use ` queryContent('articles').only(['_path','title']).find()` instead
